### PR TITLE
Update lib/pdk/module/metadata.rb with latest OS versions / Archlinux

### PR DESCRIPTION
**Pull request by bastelfreak**
This is based on a Twitter conversation: https://twitter.com/dev_el_ops/status/1032895068707930112

### CODE REVIEW

File: lib/pdk/module/metadata.rb [hunk 1]
```diff
@@ -54,12 +54,16 @@ class Metadata
...
+        'Archlinux' => {
+          'operatingsystem'        => 'Archlinux'
+        }
       }.freeze
 
       DEFAULT_OPERATING_SYSTEMS = [
         'RedHat based Linux',
         'Debian based Linux',
         'Windows',
+        'Archlinux',
```
Comment (scotje) [inline lib/pdk/module/metadata.rb]: I'd like to leave the default list as is for now, in the near future you will be able to customize this with local config.

File: lib/pdk/module/metadata.rb [hunk 2]
```diff
@@ -54,12 +54,16 @@ class Metadata
           'operatingsystem'        => 'windows',
           'operatingsystemrelease' => ['2008 R2', '2012 R2', '10'],
         },
+        'Archlinux' => {
```
Comment (scotje) [inline lib/pdk/module/metadata.rb]: The user friendly name appears to be "Arch Linux" so we should use that as the key here.

File: lib/pdk/module/metadata.rb (hunk 1 above)
Comment (bastelfreak) [reply lib/pdk/module/metadata.rb]: Well I had to try to sneak it in :D

**Comment by scotje** [comment]
@bastelfreak thanks for the contribution to PDK!

Looks like the CI failures are due to some hard coded expectations in the unit tests around what the OS support metadata should look like. I can probably take a look at those next week if needed.

File: lib/pdk/module/metadata.rb (hunk 2 above)
Comment (bastelfreak) [reply lib/pdk/module/metadata.rb]: updated

**Comment by bastelfreak** [comment]
can you recheck? You should already have write perms to my feature branch.

**Comment by scotje** [comment]
Oops, OK I'll see what I can do :)

**Comment by scotje** [comment]
OK, hopefully that commit fixes up the unit test failures.

**Comment by SimonHoenscheid** [comment]
Windows checks are failing, but it seems like test preconditions are not met. @scotje any idea how to fix this?

**Comment by scotje** [comment]
We are going to take a pass at updating the defaults here as part of PDK-1366 which will likely integrate many of these changes.

**Pull request closed without merging**